nncf/common/compression.py

```diff
@@ -10,20 +10,56 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 """
-from typing import Any
-from typing import List
-from typing import Optional
-from typing import Tuple
-from typing import TypeVar
+from typing import Dict, Optional, List, Tuple, Any, TypeVar
 
+from nncf import NNCFConfig
+from nncf.api.compression import CompressionAlgorithmBuilder
 from nncf.api.compression import CompressionAlgorithmController
 from nncf.common.schedulers import StubCompressionScheduler
 from nncf.common.utils.backend import BackendType
 from nncf.common.utils.backend import infer_backend_from_model
+from nncf.config.extractors import extract_algo_specific_config
 
 ModelType = TypeVar('ModelType')
 
 
+class BaseCompressionAlgorithmBuilder(CompressionAlgorithmBuilder):
+    _registered_name = None  # Attribute will set by COMPRESSION_ALGORITHMS registry
+
+    def __init__(self, config: NNCFConfig, should_init: bool = True):
+        """
+        Initializes internal state of the compression algorithm builder
+
+        :param config: The top-level NNCFConfig object (i.e. parsed from a .json and extended with
+            all necessary objects required for compression such as initialization data loaders).
+        :param should_init: If False, trainable parameter initialization will be
+            skipped during building.
+        """
+        super().__init__()
+        self.config = config
+        self.should_init = should_init
+        self._algo_config = self._get_algo_specific_config_section()
```

Comment:
```suggestion
        self._algo_config = extract_algo_specific_config(self.config, self._registered_name)
```

Simple is better than complex. This change will remove many lines from the code.

Reply:
Done

Reply:
Precommit failures reminded me exactly why I had to take the approach with a separate method to get algo specific config. Some builders (such as composite, experimental, tensor stat collection) do not have a registered name at all and do not appear in configs, so they don't need to access the config and there should be an option to override the algo specific config collection in the corresponding class.

Reply:
@alexsu52

Reply:
Thank you for highlighting this. It looks like WA. Please, consider removing this code from the base builder since it is not common to every CompressionAlgorithmBuilder. It's not the stopper for merging PR.